File: Solution1773.java
class Solution {
    public int countMatches(List<List<String>> items, String ruleKey, String ruleValue) {
        int itemIndex = getItemIndex(ruleKey);

        int result = 0;
        for (List<String> item : items) {
            String itemValue = item.get(itemIndex);
            if (itemValue.equals(ruleValue)) result++;
        }
        return result;
    }

    private int getItemIndex(String ruleKey) {
        int index = -1;
        if (ruleKey.equals("type")) {
            index = 0;
        } else if (ruleKey.equals("color")) {
            index = 1;
        } else if (ruleKey.equals("name")) {
            index = 2;
        }
        return index;
    }
}

File: rust/src/solution1773.rs
struct Solution {}

impl Solution {
    pub fn count_matches(items: Vec<Vec<String>>, rule_key: String, rule_value: String) -> i32 {
        let index = Solution::get_index(&rule_key) as usize;

        let mut result = 0;
        for item in items {
            if item.get(index).expect("invalid index") != &rule_value { continue }
            result += 1;
        }
        return result;
    }

    fn get_index(rule_key: &str) -> i32 {
        return  match rule_key {
            "type" => 0,
            "color" => 1,
            "name" => 2,
            _ => -1,
        };
    }
}